specs/altair/light-client/p2p-interface.md

```diff
@@ -0,0 +1,257 @@
+
+For full nodes, the following validations MUST additionally pass before forwarding the `finality_update` on the network.
+- _[IGNORE]_ The received `finality_update` matches the locally computed one exactly (as defined in [`create_light_client_finality_update`](./full-node.md#create_light_client_finality_update))
+
+For light clients, the following validations MUST additionally pass before forwarding the `finality_update` on the network.
+- _[REJECT]_ The `finality_update` is valid -- i.e. validate that `process_light_client_finality_update` does not indicate errors
+- _[IGNORE]_ The `finality_update` advances the `finalized_header` of the local `LightClientStore` -- i.e. validate that processing `finality_update` increases `store.finalized_header.slot`
+
+Light clients SHOULD call `process_light_client_finality_update` even if the message is ignored.
+
+###### `light_client_optimistic_update`
+
+This topic is used to propagate the latest `LightClientOptimisticUpdate` to light clients, allowing them to keep track of the latest `optimistic_header`.
+
+The following validations MUST pass before forwarding the `optimistic_update` on the network.
+- _[IGNORE]_ No other `optimistic_update` with a lower or equal `attested_header.slot` was already forwarded on the network
+- _[IGNORE]_ The `optimistic_update` is received after the block at `signature_slot` was given enough time to propagate through the network -- i.e. validate that one-third of `optimistic_update.signature_slot` has transpired (`SECONDS_PER_SLOT / INTERVALS_PER_SLOT` seconds after the start of the slot, with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance)
+
+For full nodes, the following validations MUST additionally pass before forwarding the `optimistic_update` on the network.
+- _[IGNORE]_ The received `optimistic_update` matches the locally computed one exactly (as defined in [`create_light_client_optimistic_update`](./full-node.md#create_light_client_optimistic_update))
+
+For light clients, the following validations MUST additionally pass before forwarding the `optimistic_update` on the network.
+- _[REJECT]_ The `optimistic_update` is valid -- i.e. validate that `process_light_client_optimistic_update` does not indicate errors
+- _[IGNORE]_ The `optimistic_update` either matches corresponding fields of the most recently forwarded `LightClientFinalityUpdate` (if any), or it advances the `optimistic_header` of the local `LightClientStore` -- i.e. validate that processing `optimistic_update` increases `store.optimistic_header.slot`
+
+Light clients SHOULD call `process_light_client_optimistic_update` even if the message is ignored.
+
+### The Req/Resp domain
+
+#### Messages
+
+##### GetLightClientBootstrap
+
+**Protocol ID:** `/eth2/beacon_chain/req/light_client_bootstrap/1/`
```

Comment:
Curious, should it be `beacon_chain/` or `light_client/`?
I see that it's a "beacon chain" light client, but for implementers & service providers, maybe there some benefits of distinguishing them?

@etan-status what do you think?

Reply:
I don't have a strong opinion on this.

`beacon_chain` fits for these reasons:
- What is being downloaded are partial beacon blocks / beacon states
- The other light client related messages, i.e., sync contributions, are also rooted in `beacon_chain`
- The light client still needs to deal with the `status`, `ping`, `metadata`, `goodbye` messages, so it cannot solely use `light_client`, at least on libp2p

`light_client` fits for these reasons:
- The topics are not necessary for collecting validator rewards
- Easier to reason starting point to direct a client implementation to

Other impacted API should also be considered, e.g., REST, should it have it under the light client umbrella, or the beacon chain one? What about portal?

Reply:
I'm fine with either way too! Let's keep it to `beacon_chain` now unless someone has a strong objection one day.